src/variable.js

```diff
@@ -40,20 +40,25 @@ function replaceListDelimiters(varValue, varName = '') {
  * @returns {String} Converted value
  */
 function resolveEnvVars(varValue) {
-  const envUnixRegex = /(\\*)(\$(\w+)|\${(\w+)})/g // $my_var or ${my_var} or \$my_var
-  return varValue.replace(
-    envUnixRegex,
-    (_, escapeChars, varNameWithDollarSign, varName, altVarName) => {
-      // do not replace things preceded by a odd number of \
-      if (escapeChars.length % 2 === 1) {
-        return varNameWithDollarSign
-      }
-      return (
-        escapeChars.substr(0, escapeChars.length / 2) +
-        (process.env[varName || altVarName] || '')
-      )
-    },
-  )
+  const envUnixRegex = /(\\*)(\$(\w+)|\${(\w+)(?::([^}]+))?})/g // $my_var or ${my_var} or \$my_var
+  return varValue.replace(envUnixRegex, (...match) => {
+    const [
+      ,
+      escapeChars,
+      varNameWithDollarSign,
+      varName,
+      altVarName,
+      defaultValue = '',
+    ] = match
```

Comment:
Used destructuring instead of arguments to avoid eslint error (function > 6 params), I wasn't sure if a ``// eslint-disable-line`` was acceptable here